CMakeLists.txt

```diff
@@ -25,6 +25,10 @@ endif()
 
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake/modules/")
 
+if(${CMAKE_BUILD_TYPE} MATCHES "Debug")
+  add_definitions(-DDEBUG)
```

Comment:
@liupan1111 
Please wrap this somewhere in a GCC ifdef-block that does this "weird"  stuff.
The way to get debuggable code in Cmake is following the Cmake setup for that.

If you'd want to debug why not use is like it is done in the ./do_freebsd.sh file by setting 
the flags calling cmake. Which is what you now sort of do in an awkward way.
```
rm -rf build && ./do_cmake.sh "$*" \
        -D CMAKE_BUILD_TYPE=Debug \
        -D CMAKE_CXX_FLAGS_DEBUG="$CXX_FLAGS_DEBUG -O0 -g" \
        -D CMAKE_C_FLAGS_DEBUG="$C_FLAGS_DEBUG -O0 -g" \
```
And I get perfectly debuggable stuff.